services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -17,7 +17,15 @@ class ProjectService(projectRepository: ProjectRepository, projectVersioning: Pr
     projectRepository.getProjectById(projectId).flatMap(getForUserOrFail(_, userId))
 
   def getUserProjectByName(namePattern: String, userId: UserId): Future[Project] =
-    projectRepository.getProjectByName(namePattern).flatMap(getForUserOrFail(_, userId))
+    projectRepository
+      .getProjectsByName(namePattern)
+      .flatMap(
+        projects =>
+          projects.find(_.ownerId == userId) match {
+            case Some(project) => Future.successful(project)
+            case None          => Future.failed(new ProjectNotFoundException)
+          }
+      )
 
   private def getForUserOrFail(project: Option[Project], userId: UserId): Future[Project] =
```

Comment:
I think you must find a way to reuse this method. If we accept your changes, then we have two places that handle project visibility and they _already_ handle it differently.

Reply:
I've returned getForUserOrFail for this

Reply:
But it is the same thing as before.
You must maintain only one place that decides whether the user can see the project or not. Now we have two 
`projects => getForUserOrFail(projects.find(_.ownerId == userId), userId)`
`case Some(project) if project.ownerId == userId => Future.successful(project)`
This one place might accept several projects instead of one and then decide what it should return. It shouldn't be a problem. Remember, `Option[T]` is basically a collection of `T`s that can contain at most 1 element.